1.5/Source/PawnStorages/PawnStorages/Farm/Comps/CompFarmStorage.cs

```diff
@@ -24,9 +24,35 @@ public class CompFarmStorage : CompPawnStorage
         public float NutritionRequiredPerDay() => compAssignable.AssignedPawns.Sum(animal =>
             SimplifiedPastureNutritionSimulator.NutritionConsumedPerDay(animal.def, animal.ageTracker.CurLifeStage));
 
-        public new void StorePawn(Pawn pawn)
+        public IEnumerable<PawnKindDef> HeldPawnTypes => this.innerContainer.innerList.Select(p => p.kindDef).Distinct();
+
+        public Dialog_AutoSlaughter.AnimalCountRecord CountForType(ThingDef def)
+        {
+            List<Pawn> pawns = innerContainer.innerList.Where(p => p.kindDef.race == def).ToList();
+            int total = pawns.Count;
+            int male = pawns.Count(p => p.gender == Gender.Male && p.ageTracker.Adult);
+            int maleYoung = pawns.Count(p => p.gender == Gender.Male && !p.ageTracker.Adult);
+            int female = pawns.Count(p => p.gender == Gender.Female && p.ageTracker.Adult);
+            int femaleYoung = pawns.Count(p => p.gender == Gender.Female && !p.ageTracker.Adult);
```

Comment:
can this not be replaced by your CountForType util

Reply:
This _is_ the `CountForType` util :)